readthedocs/projects/models.py

```diff
@@ -925,6 +929,21 @@ class Meta(object):
 
     objects = HTMLFileManager()
 
+    @cached_property
+    def json_file_path(self):
+        basename = os.path.splitext(self.name)[0]
+        full_path = self.project.get_production_media_path(type_='json',
+                                                           version_slug=self.version.slug,
+                                                           include_file=False)
+
+        file_path = find_file(basename=basename, pattern='*.fjson', path=full_path)
```

Comment:
I don't think we need to do all this I/O. Shouldn't it simply be:

`os.path.join(full_path, self.path.replace('.html', '.fjson')` or similar? It should be in the same path structure as the existing `path`.